### PR TITLE
[11.0][NEW] website_sale_product_attribute_value_filter_existing

### PR DESCRIPTION
**Pull request by sergio-teruel**
Display only attribute values which exists in /shop after a search is complete
cc @Tecnativa

### CODE REVIEW

File: website_sale_product_attribute_value_filter_existing/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    'name': 'Website Sale Attribute Value Existing',
+    'summary': 'Allow hide attributes values not used in variants',
+    'version': '11.0.1.0.0',
+    'development_status': 'Beta',
+    'category': 'Website',
+    'website': 'https://github.com/OCA/e-commerce',
+    'author': 'Tecnativa, Odoo Community Association (OCA)',
+    'license': 'AGPL-3',
```
Comment (pedrobaeza) [inline website_sale_product_attribute_value_filter_existing/__manifest__.py]: For website, it should LGPL

File: website_sale_product_attribute_value_filter_existing/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            **post)
+        products = response.qcontext['products']
+        if products:
+            domain = request.env.context.get('shop_search_domain', [])
+            # get all products without limit pagination
+            products = request.env['product.template'].search(
+                domain, limit=False)
+        # Load all products without limit for the filter check on attribute
+        # values
+        response.qcontext['variants'] = products.mapped('product_variant_ids')
```
Comment (pedrobaeza) [inline website_sale_product_attribute_value_filter_existing/controllers/main.py]: Shouldn't you save directly attributes - maybe with `read` or `read_group` - for avoiding excessive prefetching?

Comment (pedrobaeza) [reply website_sale_product_attribute_value_filter_existing/controllers/main.py]: I'm seeing that this is not under your control as this is blindly passes to upstream. A comment about that is appreciated.

**Review by pedrobaeza — APPROVED** [review]

**Review by Tardo — APPROVED** [review]

File: website_sale_product_attribute_value_filter_existing/__manifest__.py (hunk 1 above)
Comment (pedrobaeza) [reply website_sale_product_attribute_value_filter_existing/__manifest__.py]: @sergio-teruel please change this

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by chienandalu — APPROVED** [review]
Tested on runbot :+1:

**Comment by pedrobaeza** [comment]
The license change to LGPL is still pending

**Comment by sergio-teruel** [comment]
License changed

**Pull request merged**